Public/Src/FrontEnd/TypeScript.Net/TypeScript.Net/TypeChecking/Checker.cs

```diff
@@ -15866,6 +15914,54 @@ private IType CheckConditionalExpression(IConditionalExpression node, ITypeMappe
             return GetUnionType(new List<IType> { type1, type2 });
         }
 
+        private IType CheckSwitchExpression(ISwitchExpression node, ITypeMapper contextualMapper = null)
+        {
+            var expressionType = CheckExpression(node.Expression, contextualMapper);
+            if (!IsValidSwitchExpressionType(expressionType))
+            {
+                Error(
+                    node.Expression,
+                    Errors.Switch_expression_must_be_of_type_string,
+                    TypeToString(expressionType));
+            }
+
+            if (node.Clauses.Count == 0)
+            {
+                Error(
+                    node.Expression,
+                    Errors.Switch_expression_must_have_at_least_one_clause);
+            }
+
+            return GetUnionType(
+                node.Clauses.Select(clause => CheckExpression(clause.Expression, contextualMapper)).ToList()
+            );
+        }
+
+        private bool IsValidSwitchExpressionType(IType type)
+        {
+            var intrinsicType = type.As<IIntrinsicType>();
+            if (intrinsicType == m_stringType
+                || intrinsicType == m_numberType

```

Comment:
>   || intrinsicType == m_numberType [](start = 13, length = 35)

I thought you were constraining it to string only?

Reply:
Number can be usefull too, was too easy to add to not to do it.

---
In reply to: [274109520](https://github.com/Microsoft/BuildXL/pull/119#discussion_r274109520) [](ancestors = 274109520)

Reply:
Agree, but I think your error messages still say 'string' only

---
In reply to: [274115124](https://github.com/Microsoft/BuildXL/pull/119#discussion_r274115124) [](ancestors = 274115124,274109520)